front/lib/api/assistant/pubsub.ts

```diff
@@ -62,7 +64,7 @@ export async function editUserMessageWithPubSub(
     content: string;
     mentions: MentionType[];
   }
-): Promise<UserMessageType> {
+): Promise<Result<UserMessageType, APIErrorWithStatusCode>> {
```

Comment:
Just checking here : pubsub seems to be on the internal lib level, not on the API route level => shouldn't APIError be done by API-dedicated consumers of pubsub?

Reply:
if it's just a semantic thing, and we call APIError our generic way of handling typed error throws in JS => in that case I'd advocate for a renaming, I can do it in another PR if needed

Reply:
probably not critical though I reckon

Reply:
Yeah I hesitated creating a new Error type for that and decided that pubsub was close enough to API routes. Happy to create a PubSubError {code / message}... but then we loose the ability to decide which status code to return or the surface here is quite large and depends on the event and error case.

I can do a type equality at the beginning of the file to untight it a bit.